### PR TITLE
cleanup unused TopicFieldSchema-related code

### LINKED ISSUE #171: Remove TopicFieldSchemaSource* related code
`TopicFieldSchemaSourceProvider` and `TopicFieldSchemaSourceRepository` are obsolete now that we have a proper Zookeeper-based implementation. We can remove this implementation together with `messageSource` field in `Topic` to clean things up.

Requires that we have migrated our existing schemas to a new destination.

### PR DESCRIPTION
**Pull request by adamdubiel**
Resolves #171

### CODE REVIEW

File: integration/src/integration/java/pl/allegro/tech/hermes/integration/FilteringAvroTest.java [hunk 1]
```diff
@@ -30,11 +29,9 @@
             new MessageFilterSpecification(of("type", "avropath", "path", ".favoriteColor", "matcher", "grey"));
 
 
-    final static AvroUser BOB = new AvroUser("Bob", 50, "blue");
-    final static AvroUser ALICE = new AvroUser("Alice", 20, "magenta");
-    final static AvroUser BOB_GREY = new AvroUser("Bob", 50, "grey");
-
-    private final static SubscriptionPolicy SUBSCRIPTION_POLICY = new SubscriptionPolicy(100, 10, 1000, false, 100, 100);
+    private final static AvroUser BOB = new AvroUser("Bob", 50, "blue");
```
Comment (sputnikci) [inline integration/src/integration/java/pl/allegro/tech/hermes/integration/FilteringAvroTest.java]: \[Checkstyle\] ERROR: 'static' modifier out of order with the JLS suggestions\.

File: integration/src/integration/java/pl/allegro/tech/hermes/integration/FilteringAvroTest.java [hunk 2]
```diff
@@ -30,11 +29,9 @@
...
+    private final static AvroUser ALICE = new AvroUser("Alice", 20, "magenta");
```
Comment (sputnikci) [inline integration/src/integration/java/pl/allegro/tech/hermes/integration/FilteringAvroTest.java]: \[Checkstyle\] ERROR: 'static' modifier out of order with the JLS suggestions\.

File: integration/src/integration/java/pl/allegro/tech/hermes/integration/FilteringAvroTest.java [hunk 3]
```diff
@@ -30,11 +29,9 @@
...
+    private final static AvroUser BOB_GREY = new AvroUser("Bob", 50, "grey");
```
Comment (sputnikci) [inline integration/src/integration/java/pl/allegro/tech/hermes/integration/FilteringAvroTest.java]: \[Checkstyle\] ERROR: 'static' modifier out of order with the JLS suggestions\.

**Comment by druminski** [comment]
👍

**Comment by pbetkier** [comment]
👍

**Pull request merged**